software/18.Kafka/README.md

```diff
@@ -0,0 +1,214 @@
+kafka_1      | [2021-08-08 14:06:44,856] DEBUG [PartitionStateMachine controllerId=1002] Started partition state machine with initial state -> HashMap() (kafka.controller.ZkPartitionStateMachine)
+kafka_1      | [2021-08-08 14:06:44,866] INFO [SocketServer listenerType=ZK_BROKER, nodeId=1002] Started data-plane acceptor and processor(s) for endpoint : ListenerName(LISTENER_IN) (kafka.network.SocketServer)
+kafka_1      | [2021-08-08 14:06:44,867] INFO [SocketServer listenerType=ZK_BROKER, nodeId=1002] Started data-plane acceptor and processor(s) for endpoint : ListenerName(LISTENER_OUT) (kafka.network.SocketServer)
+kafka_1      | [2021-08-08 14:06:44,867] INFO [SocketServer listenerType=ZK_BROKER, nodeId=1002] Started socket server acceptors and processors (kafka.network.SocketServer)
+kafka_1      | [2021-08-08 14:06:44,872] INFO [KafkaServer id=1002] started (kafka.server.KafkaServer)
+kafka_1      | [2021-08-08 14:10:13,068] DEBUG [ReplicaStateMachine controllerId=1002] Started replica state machine with initial state -> HashMap() (kafka.controller.ZkReplicaStateMachine)
+kafka_1      | [2021-08-08 14:10:13,070] DEBUG [PartitionStateMachine controllerId=1002] Started partition state machine with initial state -> HashMap() (kafka.controller.ZkPartitionStateMachine)
+kafka_1      | [2021-08-08 14:10:13,099] INFO [SocketServer listenerType=ZK_BROKER, nodeId=1002] Started data-plane acceptor and processor(s) for endpoint : ListenerName(LISTENER_IN) (kafka.network.SocketServer)
+kafka_1      | [2021-08-08 14:10:13,100] INFO [SocketServer listenerType=ZK_BROKER, nodeId=1002] Started data-plane acceptor and processor(s) for endpoint : ListenerName(LISTENER_OUT) (kafka.network.SocketServer)
+kafka_1      | [2021-08-08 14:10:13,100] INFO [SocketServer listenerType=ZK_BROKER, nodeId=1002] Started socket server acceptors and processors (kafka.network.SocketServer)
+kafka_1      | [2021-08-08 14:10:13,104] INFO [KafkaServer id=1002] started (kafka.server.KafkaServer)
+```
+
+## Step 2 - Creating the producer
+
+### :bookmark_tabs: **Description**:
+In this step, you will create a service responsible for publishing messages into Kafka.
+
+You'll use the go programming language, and a package - [sarama](https://github.com/Shopify/sarama) - to create the producer.
+
+### :pushpin: **Tasks**:
+- Create a `client` folder and jump into it
+- Initialize the go module with the following command : `go mod init waiter`
+- Install sarama with `go get github.com/Shopify/sarama`
+- Create a `main.go` file
+- Create 3 functions:
+  - The `main` one
+  - `createProducer` which will create the producer and connect to the Kafka broker
+  - `publishOrder` which will publish a pizza order
+- The message that the producer will publish must be of the type `Order`, a structure containing a string - the pizza 
```

Comment:
Maybe you can add a `size` field which is an enumeration of `S, M, L, XL` to add fun
```suggestion
- The message that the producer will publish must be of the type `Order`, a structure containing a `string` - the pizza 
```